code/game/machinery/atmoalter/gas_mine.dm

```diff
@@ -111,138 +163,42 @@
-	var/environment_pressure = environment.return_pressure()
-
-	pumping.copy_from(air_contents)
-
-	var/pressure_delta = 10000
-
-	// External pressure bound
-	pressure_delta = min(pressure_delta, (max_external_pressure - environment_pressure))
-
-	// Internal pressure bound (screwed up calc, won't be used anyway)
-	//pressure_delta = min(pressure_delta, (internal_pressure - environment_pressure))
-
-	if(pressure_delta > 0.1)
-		var/transfer_moles = pressure_delta * CELL_VOLUME / (pumping.temperature * R_IDEAL_GAS_EQUATION)
-
-		var/datum/gas_mixture/removed = pumping.remove(transfer_moles)
-
-		loc.assume_air(removed)
-
-//Controls how fast gas comes out (in total)
-/obj/machinery/atmospherics/miner/proc/AirRate()
-  return internal_pressure * air_contents.volume / (R_IDEAL_GAS_EQUATION * air_contents.temperature)
+	if(has_external_power())
+		var/extra_mined_gas = Ceiling(WATT_TO_KPA_COEFFICIENT * power_load) //in KPa per tick
+		set_rate(base_gas_production + extra_mined_gas)	
+		
+		
+	tranfer_gas()
 
 
 /obj/machinery/atmospherics/miner/sleeping_agent
 	name = "\improper N2O Gas Miner"
 	overlay_color = "#FFCCCC"
-
-/obj/machinery/atmospherics/miner/sleeping_agent/AddAir()
-	var/rate = AirRate()
-	air_contents.adjust_multi(GAS_SLEEPING, rate)
-
+	gases = list(GAS_SLEEPING = 1)
 
 /obj/machinery/atmospherics/miner/nitrogen
 	name = "\improper N2 Gas Miner"
 	overlay_color = "#CCFFCC"
-
-/obj/machinery/atmospherics/miner/nitrogen/AddAir()
-	var/rate = AirRate()
-	air_contents.adjust_multi(GAS_NITROGEN, rate)
-
+	gases = list(GAS_NITROGEN = 1)
 
 /obj/machinery/atmospherics/miner/oxygen
 	name = "\improper O2 Gas Miner"
 	overlay_color = "#007FFF"
-
-/obj/machinery/atmospherics/miner/oxygen/AddAir()
-	var/rate = AirRate()
-	air_contents.adjust_multi(GAS_OXYGEN, rate)
-
+	gases = list(GAS_OXYGEN = 1)
 
 /obj/machinery/atmospherics/miner/toxins
 	name = "\improper Plasma Gas Miner"
 	overlay_color = "#FF0000"
-
-/obj/machinery/atmospherics/miner/toxins/AddAir()
-	var/rate = AirRate()
-	air_contents.adjust_multi(GAS_PLASMA, rate)
-
+	gases = list(GAS_PLASMA = 1)
 
 /obj/machinery/atmospherics/miner/carbon_dioxide
 	name = "\improper CO2 Gas Miner"
 	overlay_color = "#CDCDCD"
-
-/obj/machinery/atmospherics/miner/carbon_dioxide/AddAir()
-	var/rate = AirRate()
-	air_contents.adjust_multi(GAS_CARBON, rate)
-
+	gases = list(GAS_CARBON = 1)
 
 /obj/machinery/atmospherics/miner/air
 	name = "\improper Air Miner"
 	desc = "You fucking <em>cheater</em>."
 	overlay_color = "#70DBDB"
-
-	on = 0
-
-/obj/machinery/atmospherics/miner/air/AddAir()
-	var/rate = AirRate()
-	air_contents.adjust_multi(GAS_OXYGEN, 0.2*rate,
-	GAS_NITROGEN, 0.8*rate)
-
-
-/obj/machinery/atmospherics/miner/gas_giant
-	name = "\improper Gas Miner"
-
-/obj/machinery/atmospherics/miner/gas_giant/initialize()
-	..()
-	AddAir()
-
-/obj/machinery/atmospherics/miner/gas_giant/AddAir()
-	if(ticker)
-		air_contents.copy_from(gas_giant.GM)
-
-
-/obj/machinery/atmospherics/miner/mixed_nitrogen
```

Comment:
#33720 contains the mixed_nitrogen mixed_oxygen code

Reply:
oh right i just noticed half the file at the end is removed